### DISCUSSION TITLE
Increase in Size ?

### DISCUSSION THREAD

**Discussion by varbhat** [anacrolix/torrent#665] [General]
@anacrolix ,

I know that @YenForYang is doing amazing optimization stuff. But, i feel that size of library has increased.

Compare

https://github.com/varbhat/exatorrent/releases/tag/v0.0.3

vs

https://github.com/varbhat/exatorrent/releases/tag/v0.0.2

Size has been increased ~+2MB .What i did between version update was updating dependencies.

Not a problem,but would like to know the cause. Is it because of inlining loop?

**Comment by anacrolix** [comment]
That is interesting, did you change Go versions? Could any other modules be contributing? Do you have some insight @YenForYang ?

**Comment by varbhat** [comment]
Go version only changed from v1.17.0 to v1.17.1 (minor release) but i don't think this has affected.

Also, exatorrent has not many dependencies. You can see what things were updated here:
https://github.com/varbhat/exatorrent/commit/6b30f372b6682ecb737ced3a164b1e4af911b898

**Reply by anacrolix** [reply]
I think if you could find a tool to analyze the contributions of different modules/packages to your final binary size that would be useful. It looks like anacrolix/torrent updated a lot of depenendencies, and span out a new module in the diff you linked to. I can't spot anything specific that would have blown things out.

**Comment by bsergean** [comment]
The binary size increase could be because there is more inlining now.

https://dave.cheney.net/2020/04/25/inlining-optimisations-in-go
